src/rewriter.rs

```diff
@@ -522,13 +535,33 @@ impl Rewriter for HrefRewriter {
     fn rewrite<B>(&self, request: Request<B>) -> Result<Request<B>, RewriteError> {
         let (mut parts, body) = request.into_parts();
 
-        let uri_str = parts.uri.to_string();
-        let new_uri_str = self.pattern.replace(&uri_str, &self.replacement);
+        // Use the full URI for pattern matching
+        // This allows patterns to match both full URLs and path-only URIs
+        let input = parts.uri.to_string();
 
-        if new_uri_str != uri_str {
-            parts.uri = new_uri_str
+        let replaced = self.pattern.replace(&input, &self.replacement);
+        if replaced != input {
+            // Parse the result URI
+            let new_uri: http::Uri = replaced
+                .as_ref()
                 .parse()
-                .map_err(|_| RewriteError("Invalid URI after rewrite".to_string()))?;
+                .map_err(|_| RewriteError("Invalid URI after href rewrite".to_string()))?;
+
+            // If the result doesn't have a scheme/authority but the original did,
+            // preserve the original scheme and authority
+            let final_uri = if new_uri.scheme().is_none() && parts.uri.scheme().is_some() {
```

Comment:
I'd write it as follow to access `parts.uri.scheme()` once - but probably it's not relevant

```
let final_uri = if new_uri.scheme().is_none() {
    if let Some(scheme) = parts.uri.scheme() {
```